### PR TITLE
yabai: 3.3.10 -> 4.0.2

### PR DESCRIPTION
**Pull request by epetousis**
###### Description of changes

Bump yabai version to 4.0.2.

This was done by switching to the binary releases of yabai. As discussed in #174842, yabai requires entitlements and is subsequently code signed, which rules out pure builds from source at the moment.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/yabai/default.nix [hunk 1]
```diff
@@ -1,25 +1,19 @@
-{ lib, stdenv, fetchFromGitHub, darwin, xxd }:
+{ lib, stdenv }:
+
+ /*
+ This cannot be built from source as it requires entitlements and
+ for that it needs to be code signed.
+ */
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/yabai/default.nix:7]:
```suggestion
```

File: pkgs/os-specific/darwin/yabai/default.nix [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
-  version = "3.3.10";
+  version = "4.0.2";
 
-  src = fetchFromGitHub {
-    owner = "koekeishiya";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-8O6//T894C32Pba3F2Z84Z6VWeCXlwml3xsXoIZGqL0=";
+  src = fetchTarball {
+    url = "https://github.com/koekeishiya/${pname}/releases/download/v${version}/${pname}-v${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/yabai/default.nix:13]:
```suggestion
  // This cannot be built from source as it requires entitlements and for that it needs to be code signed.
  src = fetchzip {
    url = "https://github.com/koekeishiya/yabai/releases/download/v${version}/yabai-v${version}.tar.gz";
```

File: pkgs/os-specific/darwin/yabai/default.nix [hunk 3]
```diff
@@ -1,25 +1,19 @@
-{ lib, stdenv, fetchFromGitHub, darwin, xxd }:
+{ lib, stdenv }:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/yabai/default.nix:1]:
```suggestion
{ lib, stdenv, fetchzip }:
```

File: pkgs/os-specific/darwin/yabai/default.nix [hunk 4]
```diff
@@ -1,25 +1,19 @@
...
-  buildInputs = with darwin.apple_sdk.frameworks; [
-    Carbon
-    Cocoa
-    ScriptingBridge
-    SkyLight
-  ];
-
   installPhase = ''
     mkdir -p $out/bin
     mkdir -p $out/share/man/man1/
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/yabai/default.nix:19]: meta.description shhould not be a multiline string.

File: pkgs/os-specific/darwin/yabai/default.nix [hunk 5]
```diff
@@ -1,25 +1,19 @@
...
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-8O6//T894C32Pba3F2Z84Z6VWeCXlwml3xsXoIZGqL0=";
+  src = fetchTarball {
+    url = "https://github.com/koekeishiya/${pname}/releases/download/v${version}/${pname}-v${version}.tar.gz";
+    sha256 = "00nxzk1g0hd8jqd1r0aig6wdsbpk60551qxnvvqb9475i8qbzjf6";
   };
 
-  nativeBuildInputs = [ xxd ];
-
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/yabai/default.nix:15]: We do not need to do any patching?

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/os-specific/darwin/yabai/default.nix (hunk 5 above)
Comment (epetousis) [reply pkgs/os-specific/darwin/yabai/default.nix:15]: From what I can see, yabai doesn't appear to have any problematic references to non-writeable nix directories or nix stores, and doesn't appear to have any dependencies that need to be referenced in the derivation.

**Comment by epetousis** [comment]
Have added the requested changes.

**Comment by ryanorendorff** [comment]
It seems that PR #188322 is able to compile from source (I am using it on an Intel Mac without code signing problems), but does not work on ARM processors. Would there be a way to get the best of both worlds (source build and ARM support)?

**Comment by epetousis** [comment]
I'm not sure if having a Nix derivation with an x86 source build and an ARM binary is an issue or not, but it's probably the only way to go unless aarch64-darwin is to remain marked as broken. From what I could see, the scripting extension code imports ptrauth.h, which while present in Apple's LLVM, [hasn't yet been upstreamed](https://reviews.llvm.org/D112941) (and I was under the impression that nixpkgs doesn't package Apple's LLVM at all).

**Comment by epetousis** [comment]
Closed in favour of #188322.

**Pull request closed without merging**